### DISCUSSION TITLE
Wrong environment variable used in production

### DISCUSSION THREAD

**Discussion by ChrisThompsonTLDR** [vercel/next.js#53544] [Help] [answered]
### Summary

We have a monorepo with 3 applications.  One of the applications, CD, has a `.env.development` file.  A second application, SE does not have that file.  When we deploy to production, SE is using CD's `.env.development` file, even though it's not in the SE directory and SE is in the production environment.

If we remove `cd/.env.development` from the repo an redeploy SE, everything works as expected.

**Comment by Nadeera3784** [comment]
Hi, @ChrisThompsonTLDR,

Can't we keep envs like  .env.development / .env.production for each environment?

It's possible that the configuration is mistakenly referencing CD's

**Comment by csellis** [comment]
We experienced this same thing yesterday. We use the Org/Team Env Variables and had things stable for a long time like this. We use an env variable `PUBLIC_NODE_ENV` to show `staging` vs `production`. Yesterday staging was showing in production even though the shared environment was set to production for production.

**Comment by csellis** [comment]
For future @csellis, we experienced this again over the weekend of 3/12/23. `production` deployment on the Vercel side from Friday picked up `staging` credentials. We now have to check all of our deployments. This is a big pain @leerob.

**Comment by ChrisThompsonTLDR** [comment, marked as answer]
The support I received from Vercel told us to check this https://vercel.com/docs/monorepos/turborepo#handling-environment-variables

This resolved our issues.
